scripts/check_style.sh

```diff
@@ -3,7 +3,7 @@
 . scripts/report_errors.sh
 
 (
-REPO_ROOT="$(dirname "$0")"/..
+REPO_ROOT=${REPO_ROOT:-"$(dirname "$0")"/..}
 cd $REPO_ROOT
```

Comment:
This means if `REPO_ROOT` is unset, set it to `"$(dirname "$0")"/..` but I don't see why this change is required since `REPO_ROOT` is unset before that

Reply:
This is a simple way to allow `REPO_ROOT` to be set from the outside to change its behavior.  I've found that in some cases files in the various build directories or other files can get included that aren't desired.

Reply:
Can you explain this a bit more, please? Why would it be better to accept a REPO_ROOT set from outside instead of initializing it properly?

Reply:
I have had a number of problems with the check-syntax of CI. It feels to me like this script assumes an environment that appears in CI rather than a more loose kind of organization that can appear in a developer environment where there may be temporary files around and various build directories. 

I am not sure I would discribe this a "proper" initialization problem, but if it were, I'd say then the script should know about which folders are solidity code related (e.g. liblangutil, libsolidity) and which folders are not (e.g. .vscode .git ,build, .github) and not bother to look in, let alone, complain about file there. 

In the broader picture, this shell script would be better placed as a git commit hook rather than a CI test. Or a better formatter used which can just fix the code rather than complain about it. 

Here I was trying to make the situation incrementally better with minimal fuss. However if this is going to cause consternation, I can have this off to the side. 

Altruistically though I felt that if something like this is useful to me it might be useful to others in the short term. 

Perhaps I should open an issue to look at addressing the broader problem of checking style sooner and possibly fixing?

Reply:
The script assumes that it is placed at the point where it is placed. The repository root should be one directory level above the script. Do you think this is an assumption that might be wrong depending on the developer's environment? Please note that this is an honest question and I just want to learn why you are proposing this change and where it can be useful. The downside of this change is that it might lead to weird errors that are difficult to debug because the script depends more on the external environment.

Reply:
> Do you think this is an assumption that might be wrong depending on the developer's environment?

No. It was what I hoped would be a trivial non-controversial change.

Better would be to improve the script in one of the many other ways it could be improved such as: 

* by knowing which folders are part of the solidity git code. Changing this to run only on {cpp,h} files listed in `git ls` is probably better. If you'd like me attempt this script change this, I'd be happy to do. 
* using something better that encompases more of the style better and has an option to fix the code in addition or instead of to complaining about it.

Reply:
I agree that discovering cpp and h files should be improved, but I don't really see what this has to do with the path of the source folder itself.

Reply:
The hacky way I did it was to start in the folder where I knew what I changed and that folder I know was part of solidity.  I am not defending this, I am expaining it.

So would which is preferred: 

* redo using git-ls-tree
* open an issue to discuss ways to improve this script 
* both of the above

?

Reply:
This script uses `git grep`, so I don't see why `git ls-tree` should provide fewer files.

Reply:
Yes, you are correct. I weird verbose output when I ran this like looks and starts like this: 

```
$ ./scripts/check_style.sh 
Coding style error:
liblangutil/CharStream.h:60: {
Coding style error: liblangutil/ParserBase.cpp:114: _token != _value || _token == Token::EOS; liblangutil/ParserBase.cpp:115: _token = m_scanner->next()) ; Coding style error: liblangutil/ParserBase.cpp:114: _token != _value || _token == Token::EOS; liblangutil/ParserBase.cpp:115: _token = m_scanner->next()) ; Coding style error: liblangutil/ParserBase.cpp:114: _token != _value || _token == Token::EOS; liblangutil/ParserBase.cpp:115: _token = m_scanner->next()) ; Coding style error: liblangutil/ParserBase.cpp:114: _token != _value || _token == Token::EOS; liblangutil/ParserBase.cpp:115: _token = m_scanner->next()) ; Coding style error: liblangutil/ParserBase.cpp:114: _token != _value || _token == Token::EOS; liblangutil/ParserBase.cpp:115: _token = m_scanner->next()) ; Coding style error: liblangutil/ParserBase.cpp:114: _token = m_scanner->next()) ; Coding style error: liblangutil/ParserBase.cpp:114: Token _token = m_scanner->currentToken(); liblangutil/ParserBase.cpp:115: _token != _value || _token == Token::EOS; liblangutil/ParserBase.cpp:116: _token = m_scanner->next()) ; Coding style error: liblangutil/ParserBase.cpp:113: fatalParserError(string("Expected ") + tokenName(_value) + string(" but got ") + tokenName(tok) + liblangutil/ParserBase.cpp:114: "; can not find token to synchronize to."); liblangutil/ParserBase.cpp:116: parserError(string("Expected ") + tokenName(_value) + string(" but got ") + tokenName(tok) + liblangutil/ParserBase.cpp:117: "; deleted tokens to the next one."); testp0/Common.cpp:70: assertThrow( testp0/libsolidityp0/Common.h:2: This file is part of solidity. testp0/libsolidityp0/Common.h:4: solidity is free software: you can redistribute it and/or modify testp0/libsolidityp0/Common.h:5: it under the terms of the GNU General Public License as published by testp0/libsolidityp0/Common.h:6: the Free Software Foundation, either version 3 of the License, or testp0/libsolidityp0/Common.h:7: (at your option) any later version. testp0/libsolidityp0/Common.h:9: solidity is distributed in the hope that it will be useful, testp0/libsolidityp0/Common.h:10: but WITHOUT ANY WARRANTY; without even the implied warranty of testp0/libsolidityp0/Common.h:11: MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the testp0/libsolidityp0/Common.h:12: GNU General Public License for more details. testp0/libsolidityp0/Common.h:14: You should have received a copy of the GNU General Public License testp0/libsolidityp0/Common.h:15: along with solidity. If not, see <http://www.gnu.org/licenses/>. testparser/Common.cpp:70: assertThrow( testparser/libsolidity/SolidityParser.cpp:2: This file is part of solidity. testparser/libsolidity/SolidityParser.cpp:4: solidity is free software: you can redistribute it and/or modify testparser/libsolidity/SolidityParser.cpp:5: it under the terms of the GNU General Public License as published by testparser/libsolidity/SolidityParser.cpp:6: the Free Software Foundation, either version 3 of the License, or testparser/libsolidity/SolidityParser.cpp:7: (at your option) any later version ...```

and was confused. This however has nothing to do with `REPO_ROOT`.  So let me close this and think about this some more.

Sorry for the noise.